extension/user/registerUser.js

```diff
@@ -0,0 +1,55 @@
+const uuidv4 = require('uuid/v4')
+const crypto = require('crypto')
+/**
+ * @typedef {Object} RegisterInputArgs
+ * @property {string} mail
+ * @property {string} password
+ * @property {string} firstName
+ * @property {string} lastName
+ * @property {?string} gender
+ * @property {?string} birthday
+ * @property {?string} phone
+ */
+
+/**
+ * Validation and sanitation is done at previous step
+ * @param {SDKContext} context
+ * @param {RegisterInputArgs} input
+ * @param {function} cb
+ */
+module.exports = (context, input, cb) => {
+  // is logged in
+  if (context.meta.userId) {
+    return cb(null, {
+      userId: context.meta.userId
```

Comment:
Logged in users should not be given the opportunity to register (because entering the data was for nothing). Since this should be avoided in favour of usability, it should instead either result in an "invalid call" error or at least be logged as an error, so someone is at least able to notice.

Reply:
done